team_red/gen/generation_service.py

```diff
@@ -12,36 +10,17 @@
 _LOGGER = logging.getLogger(__name__)
 _LOGGER.addHandler(logging.NullHandler())
 
-
 class GenerationService:
     def __init__(self, config: GenerationConfig) -> None:
         self._config = config
-        self._pipeline: Optional[Pipeline] = None  # type: ignore[no-any-unimported]
 
-    @property
-    def pipeline(self) -> Pipeline:  # type: ignore[no-any-unimported]
-        if not self._pipeline:
-            model = AutoModelForCausalLM.from_pretrained(
-                model_path_or_repo_id=self._config.model.name,
-                model_file=self._config.model.file,
-                model_type=self._config.model.type,
-                context_length=self._config.model.context_length,
-                temperature=self._config.model.temperature,
-                repetition_penalty =self._config.model.repetition_penalty,
-                last_n_tokens = self._config.model.last_n_tokens,
-                top_k = self._config.model.top_k,
-                top_p = self._config.model.top_p,
-                hf=True,
-            )
-            tokenizer = AutoTokenizer.from_pretrained(model)
-            self._pipeline = pipeline(
-                "text-generation",
-                model=model,
-                tokenizer=tokenizer,
-                device_map="auto",
-                return_full_text=False,
-            )
-        return self._pipeline
+    def set_model(self) -> AutoModelForCausalLM:  # type: ignore[no-any-unimported]
```

Comment:
the naming could be improved. a 'setter' method usually gets some parameters to set AND sets something such as a member variable. This method does neither. `def create_model` or `init_model` would be more appropriate.

Reply:
I agree. Changed it.